### ISSUE TITLE
Craft 2-3 migration error - Console Request - Notification Email

### ISSUE THREAD

**Issue by jaydensmith** [barrelstrength/craft-sprout-forms#366]
### Description (screenshots, logs, ...)
Another migration issue caused by accessing `Craft::$app->request` from a command-line request in `NotificationEmailQuery.php`.

```
Exception: Calling unknown method: craft\console\Request::getBodyParam() (/Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/base/Component.php:300)
#0 /Users/jaydensmith/public_html/site.com/vendor/barrelstrength/sprout-base-email/src/elements/db/NotificationEmailQuery.php(44): yii\base\Component->__call('getBodyParam', Array)
#1 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1055): barrelstrength\sproutbaseemail\elements\db\NotificationEmailQuery->beforePrepare()
#2 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\mysql\QueryBuilder))
#3 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/db/Query.php(146): yii\db\QueryBuilder->build(Object(barrelstrength\sproutbaseemail\elements\db\NotificationEmailQuery))
#4 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/db/Query.php(237): yii\db\Query->createCommand(Object(craft\db\Connection))
#5 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/db/Query.php(232): yii\db\Query->all(NULL)
#6 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1205): craft\db\Query->all(NULL)
#7 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/services/Elements.php(789): craft\elements\db\ElementQuery->all()
#8 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/services/Elements.php(731): craft\services\Elements->updateDescendantSlugsAndUris(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail), true)
#9 /Users/jaydensmith/public_html/site.com/vendor/barrelstrength/sprout-base-email/src/elements/NotificationEmail.php(306): craft\services\Elements->updateElementSlugAndUri(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#10 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/services/Elements.php(522): barrelstrength\sproutbaseemail\elements\NotificationEmail->afterSave(true)
#11 /Users/jaydensmith/public_html/site.com/vendor/barrelstrength/sprout-base-email/src/services/NotificationEmails.php(50): craft\services\Elements->saveElement(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail), false)
#12 /Users/jaydensmith/public_html/site.com/vendor/barrelstrength/sprout-forms/src/migrations/m180314_161540_craft2_to_craft3.php(159): barrelstrength\sproutbaseemail\services\NotificationEmails->saveNotification(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#13 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/db/Migration.php(56): barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3->safeUp()
#14 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/db/MigrationManager.php(243): craft\db\Migration->up(true)
#15 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3))
#16 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/services/Updates.php(225): craft\db\MigrationManager->up()
#17 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/console/controllers/MigrateController.php(252): craft\services\Updates->runMigrations(Array)
#18 [internal function]: craft\console\controllers\MigrateController->actionAll()
#19 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#20 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#21 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#22 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#23 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#24 /Users/jaydensmith/public_html/site.com/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#25 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#26 /Users/jaydensmith/public_html/site.com/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#27 /Users/jaydensmith/public_html/site.com/craft(22): yii\base\Application->run()
#28 {main}
Exception 'craft\errors\MigrateException' with message 'An error occurred while migrating Sprout Forms.'
```

### Steps to reproduce

1. Migrate from 2-3 via command line `./craft migrate/all`

### Additional info

- Craft version: 3.1.30
- PHP version: 7.1
- Database driver & version:  
- Plugins & versions:

_For some issues, we may request that you send the following items to Support <sprout@barrelstrengthdesign.com>. If you feel your issue may fall in that category, feel free to send things over after you post this issue and reference the issue in your email._

- `composer.json`
- `storage/logs/web.log` (or a zip of the whole log folder)
- An export of your database (or, if you can't share the whole db, the specific Sprout tables)

**Comment by BenParizek** [comment]
@jaydensmith Can you confirm what versions of the Sprout plugins you are updating to are?

I believe this was resolved in Sprout Base Email v1.1.5

**Comment by BenParizek** [comment]
@jaydensmith If you can try updating again to the latest version, we just released Sprout Forms 3.6.0 which address a few other Craft 2 to 3 migration scenarios as well.

**Comment by jaydensmith** [comment]
Thanks @BenParizek that must be it
